pyhf/pdf.py

```diff
@@ -435,33 +436,48 @@ def expected_data(self, pars, include_auxdata=True):
         )
         return tensorlib.concatenate(tocat)
 
-    def constraint_logpdf(self, auxdata, pars):
-        normal = self.constraints_gaussian.logpdf(auxdata, pars)
-        poisson = self.constraints_poisson.logpdf(auxdata, pars)
-        return normal + poisson
-
-    def mainlogpdf(self, maindata, pars):
+    def constraint_pdf(self, pars):
+        tensorlib, _ = get_backend()
+        gview = self.constraints_gaussian.dataview()
+        pview = self.constraints_poisson.dataview()
+
+        factors = []
+        projections = []
+        if gview is not None:
+            gpdf  = self.constraints_gaussian.pdf(pars)
+            factors.append(gpdf)
+            projections.append(gview)
+        if pview is not None:
+            ppdf = self.constraints_poisson.pdf(pars)
+            factors.append(ppdf)
+            projections.append(pview)
+
+        sim = Simultaneous(factors, projections)
+        return sim
+
+    def mainpdf(self, pars):
```

Comment:
I'm realizing I don't understand what mainpdf (or mainlogpdf) was for. E.G. why only Poisson?